jax/lax/lax.py

```diff
@@ -5259,22 +5259,24 @@ def _check_shapelike(fun_name, arg_name, obj):
 
 
 def _dynamic_slice_indices(operand, start_indices):
-  if not isinstance(start_indices, (tuple, list)):
-    if start_indices.ndim != 1:
-      raise ValueError("Slice indices must be a 1D sequence, got {}"
-                       .format(start_indices.shape))
-    start_indices = [squeeze(slice(start_indices, [i], [i+1]), dimensions=(0,))
-                     for i in range(operand.ndim)]
-  else:
-    start_indices = [onp.asarray(i, dtype=dtypes.int_) if isinstance(i, int)
-                     else i for i in start_indices]
   if len(start_indices) != operand.ndim:
     msg = ("Length of slice indices must match number of operand dimensions ({} "
           "vs {})")
     raise ValueError(msg.format(len(start_indices), operand.shape))
   # map int over operand.shape to raise any dynamic-shape errors
-  return [select(lt(i, _const(i, 0)), add(i, _const(i, int(d))), i)
-          for i, d in zip(start_indices, operand.shape)]
+  map(int, operand.shape)
```

Comment:
Would you prefer
```python
for s in operand.shape:
  int(s)
```
?

Reply:
Or we could do
```python
list(map(int, operand.shape))
```

Reply:
Looks like safe_map isn't imported in lax, so this is a no-op the way I implemented it. I might just import safe_map and use it here (without over-riding the map builtin like we do elsewhere in JAX).

Reply:
Will resolve this but lmk if you reckon there's still room for improvement.

Reply:
I don't have a strong preference but another option is always to use a list comprehension.

This is tangential but I think we should always set `map, unsafe_map = safe_map, map` (a suggestion I think Dougal had recently). The Python 3 `map` is quite unsafe IMO, since I don't expect iteration to be side-effecting.